File: Documentation/5.4/Samples/python/ClientApi/Session/DeletingEntities.py
from typing import Optional, Union

from ravendb.documents.commands.batches import DeleteCommandData
from ravendb.infrastructure.orders import Employee

from examples_base import ExamplesBase


class DeletingEntities(ExamplesBase):
    def setUp(self):
        super().setUp()
        with self.embedded_server.get_document_store("DeletingEntities") as store:
            with store.open_session() as session:
                session.store(Employee(), "employees/1")
                session.save_changes()

    class Foo:
        # region deleting_1
        def delete(self, key_or_entity: Union[str, object], expected_change_vector: Optional[str] = None) -> None:
            ...

        # endregion

    def test_deleting_entities(self):
        with self.embedded_server.get_document_store("DeletingEntities") as store:
            with store.open_session() as session:
                # region deleting_2

                employee = session.load("employees/1")

                session.delete(employee)
                session.save_changes()

                # endregion

            with store.open_session() as session:
                # region deleting_3

                session.delete("employees/1")
                session.save_changes()

                # endregion

            with store.open_session() as session:
                # region deleting_4

                session.delete("employees/1")

                # endregion

            with store.open_session() as session:
                # region deleting_5

                session.advanced.defer(DeleteCommandData("employees/1", change_vector=None))

                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/DeletingEntities.php
<?php

namespace RavenDB\Samples\ClientApi\Session;

use RavenDB\Documents\Commands\Batches\DeleteCommandData;
use RavenDB\Samples\Infrastructure\DocumentStoreHolder;
use RavenDB\Samples\Infrastructure\Orders\Employee;

interface FooInterface {
    # region deleting_1
    public function delete(?object $entity): void;

    public function delete(?string $id): void;

    public function delete(?string $id, ?string $expectedChangeVector): void;
    # endregion
}

class DeletingEntities
{
    public function testSamples(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {
            $session = $store->openSession();
            try {
                # region deleting_2
                $employee = $session->load(Employee::class, "employees/1");

                $session->delete($employee);
                $session->saveChanges();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region deleting_3
                $session->delete("employees/1");
                $session->saveChanges();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region deleting_4
                $session->delete("employees/1");
                # endregion

                # region deleting_5
                $session->advanced()->defer(new DeleteCommandData("employees/1", null));
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
